### PR TITLE
docs(kubernetes): improve docstrings for methods

### PR DESCRIPTION
**Pull request by pedrooot**
### Description

This pull request introduces comprehensive documentation for the `KubernetesProvider` class in `prowler/providers/kubernetes/kubernetes_provider.py`. The changes include detailed docstrings for the class, its attributes, and its methods, providing clear descriptions, expected arguments, return values, exceptions, and usage examples.

### Documentation Enhancements:

* Added a class-level docstring to `KubernetesProvider`, detailing its attributes and methods.
* Expanded the `__init__` method docstring to include exceptions, return values, and usage examples. [[1]](diffhunk://#diff-b330be3f697f73e8884e166b2c98a24f34f5348f8eea43861a84d0c82d10d386R81) [[2]](diffhunk://#diff-b330be3f697f73e8884e166b2c98a24f34f5348f8eea43861a84d0c82d10d386R92-R126)
* Enhanced the `setup_session` method docstring with exceptions.
* Improved the `test_connection` method docstring by adding exceptions and usage examples.
* Updated docstrings for various methods (`search_and_save_roles`, `get_context_user_roles`, `get_all_namespaces`, `get_pod_current_namespace`, and `print_credentials`) to include exceptions and usage examples. [[1]](diffhunk://#diff-b330be3f697f73e8884e166b2c98a24f34f5348f8eea43861a84d0c82d10d386R397-R407) [[2]](diffhunk://#diff-b330be3f697f73e8884e166b2c98a24f34f5348f8eea43861a84d0c82d10d386R435-R440) [[3]](diffhunk://#diff-b330be3f697f73e8884e166b2c98a24f34f5348f8eea43861a84d0c82d10d386R477-R487) [[4]](diffhunk://#diff-b330be3f697f73e8884e166b2c98a24f34f5348f8eea43861a84d0c82d10d386R520-R525) [[5]](diffhunk://#diff-b330be3f697f73e8884e166b2c98a24f34f5348f8eea43861a84d0c82d10d386R541-R543)

### Checklist

- Are there new checks included in this PR? Yes / No
    - If so, do we need to update permissions for the provider? Please review this carefully.
- [x] Review if the code is being covered by tests.
- [x] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings
- [x] Review if backport is needed.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/kubernetes/kubernetes_provider.py [hunk 1]
```diff
@@ -66,6 +89,41 @@ def __init__(
...
+
+        Raises:
+            KubernetesCloudResourceManagerAPINotUsedError: If the Kubernetes Cloud Resource Manager API is not used.
+            KubernetesError: If an error occurs.
+
+        Returns:
+            KubernetesProvider: The KubernetesProvider instance.
+
+        Usage:
+            - Using the kubeconfig file and specifying the context and namespace:
```
Comment (jfagoagas) [inline prowler/providers/kubernetes/kubernetes_provider.py]: I'd add a section for authentication and then others for namespaces and context as you did with GCP, Azure and AWS.

File: prowler/providers/kubernetes/kubernetes_provider.py [hunk 2]
```diff
@@ -250,8 +313,23 @@ def test_connection(
...
             Connection: A Connection object.
+
+        Raises:
+            KubernetesInvalidKubeConfigFileError: If the kubeconfig file is invalid.
+            KubernetesInvalidProviderIdError: If the provider ID is invalid.
+            KubernetesSetUpSessionError: If an error occurs while setting up the session.
+            KubernetesAPIError: If an error occurs while testing the connection.
+
+        Usage:
+            >>> connection = KubernetesProvider.test_connection(
```
Comment (jfagoagas) [inline prowler/providers/kubernetes/kubernetes_provider.py]: I'd add an example with the path and other with the content.

**Review by jfagoagas — CHANGES_REQUESTED** [review]

File: prowler/providers/kubernetes/kubernetes_provider.py (hunk 1 above)
Comment (pedrooot) [reply prowler/providers/kubernetes/kubernetes_provider.py]: DOne

File: prowler/providers/kubernetes/kubernetes_provider.py (hunk 2 above)
Comment (pedrooot) [reply prowler/providers/kubernetes/kubernetes_provider.py]: Done

**Review by jfagoagas — APPROVED** [review]

**Pull request merged**